src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -43,24 +48,18 @@ public void setShooterSpeed(double speed) {
   }
 
   /**
-     * true opens the piston
-     * false closes the piston
-     * @param state the state of the piston
+     * Getter for the shooter's speed in rotations per minute
+     * @return the RPM of the shooter
   */
-  public void setPositionState(boolean state) {
-    if (state) {
-        ampPiston.set(DoubleSolenoid.Value.kForward); 
-    } 
-    else 
-    {
-        ampPiston.set(DoubleSolenoid.Value.kReverse); 
-    }
-  }
+  public double getShooterSpeedRPM() {
+    this.m_velocity.refresh();
```

Comment:
maybe move every `refresh()` (in every subsystem) to the subsystems `periodic()`

Reply:
MutableMeasure משתמשים בוא לעקוב אחרי מידע של סנסורים או מידות אחרות שמשתנות בזמן אמת
למרות שrefresh כל הזמן שתמיד יהיה מעודכן זה חד משמעית הכי טוב לפי מה שהבנתי עדיף לא לעשות את זה אלה אם כן חייבים בשביל הזיכרון של הrio

יכול להיות שעדיף אבל אני לא רוצה לשנות את זה עדיין במקרה שזה יעשה בעייה שהrio קורס או משהו (נראלי כדי לכם לבדוק את זה בסדנה אם יש לכם זמן)